File: Number Complement (476)/476.py
# Runtime: 20 ms
# Beats 100% of Python submissions

class Solution(object):
    def findComplement(self, num):
        """
        :type num: int
        :rtype: int
        """
        i = 0
        
        while i < num:
            i = i << 1 | 1
            
        return i ^ num

File: Number Complement (476)/476.java
// Runtime: 6 ms
// Beats 99.13% of Java submissions

class Solution {
    public int findComplement(int num) {
        int res = setBits(num);
        
        return res ^ num;
    }
    
    public static int setBits(int num) {
        int res = 0;
        
        while (num != 0) {
            res = (res << 1) | 1;
            num = num >> 1;
        }
        
        return res;
    }
}